src/Loader.php

```diff
@@ -634,6 +653,53 @@ public static function get_translation_chunk_data( $lang_dir, $domain, $locale )
 		return $combined_translation_data;
 	}
 
+	/**
+	 * Load translation strings from language packs for dynamic imports.
+	 *
+	 * This function combines JSON translation data auto-extracted by GlotPress
+	 * from Webpack-generated JS chunks into a single file. This is necessary
+	 * since the JS chunks are not known to WordPress via wp_register_script()
+	 * and wp_set_script_translations().
+	 */
+	public static function generate_translation_strings() {
+		$plugin_domain = explode( '/', plugin_basename( __FILE__ ) )[0];
+
+		$locale         = determine_locale();
+		$cache_filename = self::get_combined_translation_filename( $plugin_domain, $locale );
+		$lang_dir       = WP_LANG_DIR . '/plugins/';
+
+		// Bail early if not localized.
+		if ( 'en_US' === $locale ) {
+			return;
+		}
+
+		if ( ! function_exists( 'get_filesystem_method' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		$access_type = get_filesystem_method();
+		if ( 'direct' === $access_type ) {
+			\WP_Filesystem();
+			global $wp_filesystem;
+
+			// Get all translation chunk data combined into a single object.
```

Comment:
@ilyasfoo could we consolidate lines 686-695 here with this piece of code as it is identical?: https://github.com/woocommerce/woocommerce-admin/blob/013e03e59fc27d297d3a3f3a688fd61e3f6bdede/src/Loader.php#L744-L754

Reply:
That's a good suggestion! Done in 6b0c557